tests/docstrings_examples/DocTest.res

```diff
@@ -0,0 +1,399 @@
+  ->loop(list{})
+  ->List.toArray
+  ->Array.filter(({docstrings}) => Array.length(docstrings) > 0)
+}
+
+let getCodeBlocks = example => {
+  let rec loopEndCodeBlock = (lines, acc) => {
+    switch lines {
+    | list{hd, ...rest} =>
+      if (
+        hd
+        ->String.trim
+        ->String.endsWith("```")
+      ) {
+        acc
+      } else {
+        loopEndCodeBlock(rest, list{hd, ...acc})
+      }
+    | list{} => panic(`Failed to find end of code block for ${example.kind}: ${example.id}`)
+    }
+  }
+
+  let rec loop = (lines: list<string>, acc: list<string>) => {
+    switch lines {
+    | list{hd, ...rest} =>
+      switch hd
+      ->String.trim
+      ->String.startsWith("```res") {
+      | true =>
+        let code = loopEndCodeBlock(rest, list{})
+        loop(
+          rest,
+          list{
+            code
+            ->List.reverse
+            ->List.toArray
+            ->Array.join("\n"),
+            ...acc,
+          },
+        )
+      | false => loop(rest, acc)
+      }
+    | list{} => acc
+    }
+  }
+
+  example.docstrings
+  ->Array.reduce([], (acc, docstring) => acc->Array.concat(docstring->String.split("\n")))
+  ->List.fromArray
+  ->loop(list{})
+  ->List.toArray
+  ->Belt.Array.reverse
+  ->Array.join("\n\n")
+}
+
+let main = async () => {
+  let files = Fs.readdirSync("runtime")
+
+  let modules =
+    files
+    // Ignore Js modules and RescriptTools for now
+    ->Array.filter(f => !String.startsWith(f, "Js") && !String.startsWith(f, "RescriptTools"))
+    ->Array.filter(f => f->String.endsWith(".res") || f->String.endsWith(".resi"))
+    ->Array.reduce([], (acc, cur) => {
+      let isInterface = cur->String.endsWith(".resi")
+
+      let resi = Path.join2("runtime", cur ++ "i")
+
+      // If .resi files exists append to array
+      if !isInterface && Fs.existsSync(resi) {
+        Array.concat(acc, [cur ++ "i"])
+      } else if !Array.includes(acc, cur) {
+        Array.concat(acc, [cur])
+      } else {
+        acc
+      }
+    })
+    ->Array.map(f => extractDocFromFile(Path.join(["runtime", f]))->getExamples)
+    ->Array.flat
+
+  let compilationResults =
+    await modules
+    ->Array.map(async example => {
+      let id = example.id->String.replaceAll(".", "__")
+      let rescriptCode = example->getCodeBlocks
+      let jsCode = await compileTest(~id, ~code=rescriptCode)
+      (example, (rescriptCode, jsCode))
+    })
+    ->Promise.all
+
+  let (compiled, compilationErrors) = compilationResults->Array.reduce(([], []), (
+    acc,
+    (example, (rescriptCode, jsCode)),
+  ) => {
+    let (lhs, rhs) = acc
+    switch jsCode {
+    | Ok(jsCode) => lhs->Array.push((example, rescriptCode, jsCode))
+    | Error(err) => rhs->Array.push((example, ReScriptError(err)))
+    }
+    (lhs, rhs)
+  })
+
+  let runtimeErrors =
+    (await compiled
+    ->Array.filter((({id}, _, _)) => !Array.includes(ignoreRuntimeTests, id))
+    ->Array.map(async ((example, rescriptCode, jsCode)) => {
```

Comment:
If there is too much parallelism here, you might consider a solution like in #7081.